---
title: "gce/WARN/2025_001"
linkTitle: "WARN/2025_001"
weight: 1
type: docs
description: >
  Serial logs don't contain time synchronization errors
---

**Product**: [Compute Engine](https://cloud.google.com/compute)\
**Rule class**: WARN - Something that is possibly wrong

### Description

Time synchronization issues in GCE instances can cause various critical problems:
- Kerberos authentication failures
- TLS/SSL certificate validation errors
- Database replication problems
- Logging and monitoring inconsistencies
- Application-level errors

This rule checks for common time synchronization error patterns from NTP, Chrony,
and general system time services in the instance's serial logs.

The following error messages are detected:
- NTP errors: "time may be out of sync", "System clock is unsynchronized", "Time drift detected"
- Chrony errors: "Time offset too large", "Can't synchronise: no selectable sources"
- General errors: "Clock skew detected", "Clock skew too great"

### Remediation

1. Verify that the instance can reach Google's metadata server:

```
ping -c 3 metadata.google.internal
```

2. Check if the NTP/Chrony service is running:

```
# For systems using systemd-timesyncd
systemctl status systemd-timesyncd

# For systems using chronyd
systemctl status chronyd

# For systems using ntpd
systemctl status ntpd
```

3. Verify the time source is configured to use Google's NTP server:

```
# For Chrony
grep -E 'server|pool' /etc/chrony/chrony.conf

# For NTP
grep -E 'server|pool' /etc/ntp.conf
```

4. Ensure the configuration uses Google's recommended NTP server (metadata.google.internal):

```
# For Chrony - edit /etc/chrony/chrony.conf
server metadata.google.internal iburst

# For NTP - edit /etc/ntp.conf
server metadata.google.internal iburst
```

5. Restart the time synchronization service:

```
# For systemd-timesyncd
sudo systemctl restart systemd-timesyncd

# For Chrony
sudo systemctl restart chronyd

# For NTP
sudo systemctl restart ntpd
```

You can use the following log query to find matching log lines:

```
resource.type="gce_instance"
log_id("serialconsole.googleapis.com/serial_port_1_output")
textPayload:(
  "time may be out of sync" OR
  "System clock is unsynchronized" OR
  "Time drift detected" OR
  "no servers can be used" OR
  "time reset" OR
  "Time offset too large" OR
  "Can't synchronise" OR
  "Clock skew detected" OR
  "Clock skew too great" OR
  "Could not receive latest log timestamp"
)
```

### Further information

- [Configuring NTP on Compute Engine](https://cloud.google.com/compute/docs/instances/configure-ntp)
- [Best practices for Compute Engine](https://cloud.google.com/compute/docs/best-practices)
- [Troubleshooting SSH to Compute Engine instances](https://cloud.google.com/compute/docs/troubleshooting/troubleshooting-ssh-errors)
